alternatives.c

```diff
@@ -932,14 +932,15 @@ static int removeAll(const char * title, const char * altDir, const char * state
     struct alternativeSet set;
 
     int alt;
+    int ret_val = 0;
 
     if (readConfig(&set, title, altDir, stateDir, flags)) return 2;
 
     for (alt = 0; alt < set.numAlts; alt++) {
-        removeService(title, set.alts[alt].master.target, altDir, stateDir, flags);
+        ret_val += removeService(title, set.alts[alt].master.target, altDir, stateDir, flags);
```

Comment:
I don't think this is correct. This will save only the return code from the last call of removeService. I think it would be better to do something like ret_val = ret_val ?: removeService. Also can you squash this patch with the other two and force-push the branch?